src/values.rs

```diff
@@ -191,6 +198,28 @@ impl JitValue {
                     ptr.cast::<[u32; 8]>().as_mut().copy_from_slice(&data);
                     ptr
                 }
+                Self::BoundedInt {
+                    value,
+                    range: Range { lower, upper },
+                } => {
+                    let value = value.to_bigint();
+
+                    assert!(lower < upper, "invalid range");
+                    let prime = &PRIME.to_bigint().unwrap();
+                    let lower = lower.rem_euclid(prime);
+                    let upper = upper.rem_euclid(prime);
+                    if lower <= upper {
+                        assert!(lower <= value && value < upper);
```

Comment:
Can you change the asserts here and replace it with some return Err(..)

Reply:
done